## Example layered costmap 2d params with default values
# Copy it instead of making changes

# Costmap parameters
global_frame: /map
robot_base_frame: /base_link
transform_tolerance: 0.2
rolling_window: false
track_unknown_space: true
always_send_full_costmap: true
width: 95.75
height: 41.05
resolution: 0.05
origin_x: -47.875
origin_y: -20.525

update_frequency: 5.0
publish_frequency: 2.0

footprint:  [
[-0.220000, -0.080000],
[-0.120000, -0.160000],
[-0.109000, -0.160000],
[-0.109000, -0.195000],
[0.109000, -0.195000],
[0.109000, -0.160000],
[0.330000, -0.160000],
[0.330000, 0.160000],
[0.109000, 0.160000],
[0.109000, 0.195000],
[-0.109000, 0.195000],
[-0.109000, 0.160000],
[-0.220000, 0.100000]
]
footprint_padding: 0.01

# Define pre hydro plugins
plugins: 
  - {name: static_layer,       type: "layered_costmap_2d/StaticLayer"}
  - {name: obstacle_layer,       type: "layered_costmap_2d/ObstacleLayer"}
  - {name: inflation_layer,       type: "layered_costmap_2d/InflationLayer"}
# Voxel grid excluded in layered costmap for fuerte

static_layer:
  map_topic: /map
  first_map_only: false
  subsribe_to_updates: false
  track_unknown_space: true
  use_maximum: true # just for test else false
  lethal_cost_threshold: 100
  unknown_cost_value: -1
  trinary_costmap: true

obstacle_layer:
  transform_tolerance: 0.2  
  global_frame: /map
  robot_base_frame: /base_link
  map_type: costmap 
  track_unknown_space: true
  publish_cost_grid: true
  resolution: 0.05
  publish_voxel_map: false
  # sensors
  max_obstacle_height: 2.0
  obstacle_range: 50.0
  raytrace_range: 50.0  


  observation_sources: base_scan0 base_scan1
  base_scan0: {
      data_type: LaserScan,
      sensor_frame: /base_laser_link0,
      topic: base_scan0,
      expected_update_rate: 0.0,
      observation_persistence: 0.0,
      marking: true,
      clearing: true,
      min_obstacle_height: 0.0
  }

  base_scan1: {
      data_type: LaserScan,
      sensor_frame: /base_laser_link1,
      topic: base_scan1,
      expected_update_rate: 0.0,
      observation_persistence: 0.0,
      marking: true,
      clearing: true,      
      min_obstacle_height: 0.0
  } 

inflation_layer:
  inflation_radius: 0.50
  cost_scaling_factor: 10.0
